Mathlib/Analysis/InnerProductSpace/Projection.lean

```diff
@@ -586,6 +587,33 @@ variable (K)
 theorem orthogonalProjection_norm_le : ‖K.orthogonalProjection‖ ≤ 1 :=
   LinearMap.mkContinuous_norm_le _ (by norm_num) _
 
+/-- The orthogonal projection onto a closed subspace is norm non-increasing. -/
+theorem norm_orthogonalProjection_apply_le (v : E) :
+    ‖orthogonalProjection K v‖ ≤ ‖v‖ := by calc
```

Comment:
It would be nice to have this one too
```suggestion
theorem norm_orthogonalProjection_apply {v : E} (hv : v ∈ K) :
    ‖orthogonalProjection K v‖ = ‖v‖ :=
  congr(‖$(K.orthogonalProjection_eq_self_iff.mpr hv)‖)

/-- The orthogonal projection onto a closed subspace is norm non-increasing. -/
theorem norm_orthogonalProjection_apply_le (v : E) :
    ‖orthogonalProjection K v‖ ≤ ‖v‖ := by calc
```

Reply:
For the `norm_orthogonalProjection_apply`, it seems to me that we'd rather just rewrite `orthogonalProjection K v` to `v` in this case, instead of having an extra lemma for that rewrite after the norm has been applied. In general, I think it's preferable to avoid having a lemma `f v = f w` in the case that `v = w` (unless there are some motive-related or implicit/instance argument reasons to do so, which there aren't here). What do you think?

Reply:
Oh, now I understand that there are reasons to do so here, I will add this as you suggest!